src/dcqc/file.py

```diff
@@ -96,93 +131,187 @@ def __init__(
+            key: Metadata key name.
+
+        Raises:
+            KeyError: If the metadata key doesn't exist.
+
+        Returns:
+            The metadata value associated with the given key.
+        """
         if key not in self.metadata:
             url = self.url
             md = self.metadata
             message = f"File ({url}) does not have '{key}' in its metadata ({md})."
-            raise ValueError(message)
+            raise KeyError(message)
         return self.metadata[key]
 
     def is_url_local(self, url: Optional[str] = None) -> bool:
+        """Check whether a URL refers to a local location.
+
+        Args:
+            url: Local or remote location of a file.
+                Defaults to URL associated with file.
+
+        Returns:
+            Whether the URL refers to a local location.
+        """
         url = url or self.url
-        return self.LOCAL_REGEX.fullmatch(url) is not None
+        return self._LOCAL_REGEX.fullmatch(url) is not None
 
-    def is_file_local(self, url: Optional[str] = None) -> bool:
+    def is_file_local(self) -> bool:
+        """Check if the file (or a copy) is available locally.
+
+        Unlike :func:`~dcqc.file.File.is_url_local`, this method
+        considers if a locally staged copy is available regardless
+        of whether the URL is local or remote.
+
+        To retrieve the location of the local copy, you can use
+        :func:`~dcqc.file.File.get_local_path`.
+
+        Returns:
+            Whether the file has a copy available locally.
+        """
         return self.local_path is not None
 
     def get_local_path(self) -> Path:
+        """Retrieve the path of a local copy, if applicable.
+
+        Raises:
+            FileNotFoundError: If there is no local copy available.
+
+        Returns:
+            The path to the local copy.
+        """
         if self.local_path is None:
             message = "Local path is unavailable. Use stage() to create a local copy."
-            raise ValueError(message)
+            raise FileNotFoundError(message)
         return Path(self.local_path)
 
-    def stage(self, destination: Optional[str] = None, overwrite: bool = False) -> Path:
-        """Download remote files and copy local files.
+    def stage(
+        self,
+        destination: Optional[Path] = None,
+        overwrite: bool = False,
+    ) -> Path:
+        """Create local copy of local or remote file.
 
-        A destination is required for remote files.
+        A destination is not required for remote files; it
+        defaults to a temporary directory.
         Local files aren't moved if a destination is omitted.
 
         Args:
-            destination (Optional[str]): File or folder path
-                where to store the file. Defaults to None.
-            overwrite (bool): Whether to ignore existing file
-                at the target destination. Defaults to False.
+            destination: File or folder where to store the file.
```

Comment:
Did you create a custom template for the autoDocGenerator via vscode? https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring

Reply:
I was already using this extension, but I didn't know that it supported custom templates. Sweet! 

![image](https://user-images.githubusercontent.com/740725/214678563-b0b12421-3dc4-4931-907d-2213aee205ab.png)